File: Deepseek-Coder-33b-Instruct/stage2/data_construct_create_clause.cpp
#include <stdlib.h>
#include <iostream>
#include <openacc.h>

#ifndef T1
//T1:data construct create clause,V:2.7-3.3
int test1(){
    int err = 0;
    const int N = 100;
    int *data = new int[N];

    // Initialize data
    for(int i = 0; i < N; i++) {
        data[i] = i;
    }

    // Create data region
    #pragma acc data create(data[0:N])
    {
        // Check if data is correctly created
        for(int i = 0; i < N; i++) {
            if(data[i] != i) {
                err = 1;
                break;
            }
        }
    }

    delete[] data;

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    failed = failed + test1();
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Deepseek-Coder-33b-Instruct/stage2/data_construct_create_clause.f90
!T1:data construct create clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, PARAMETER :: N = 100
        INTEGER :: i
        INTEGER, DIMENSION(N) :: a, a_host

        ! Initialize host array
        DO i = 1, N
          a_host(i) = i
        END DO

        ! Copy data to device
        !$acc data create(a)
        !$acc update device(a_host)
        !$acc end data

        ! Copy data back to host
        !$acc data copyout(a)
        !$acc update host(a)
        !$acc end data

        ! Check results
        DO i = 1, N
          IF (a(i) .ne. a_host(i)) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM